### PR TITLE
DUOS-1746[risk=no] Update Create React App and React versions and remove React-PDF based components

### PR DESCRIPTION
**Pull request by JVThomas**
Addresses [DUOS-1746](https://broadworkbench.atlassian.net/browse/DUOS-1746)

PR updates application to use latest version of Create React App (which uses Webpack 5) and React to v18 (concurrent mode support). PR also removes `@react-pdf` as a dependency alongside the old pdf components that utilized it.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by rushtong — APPROVED** [review]
Overall, looks good! I do see a compile warning, but it seems innocuous.

```
WARNING in ./node_modules/paging-algorithm/dist/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/grushton/develop/duos-ui/node_modules/paging-algorithm/src/index.ts' file: Error: ENOENT: no such file or directory, open '/Users/grushton/develop/duos-ui/node_modules/paging-algorithm/src/index.ts'
```

**Comment by JVThomas** [comment]
> Overall, looks good! I do see a compile warning, but it seems innocuous.
> 
> ```
> WARNING in ./node_modules/paging-algorithm/dist/index.js
> Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
> Failed to parse source map from '/Users/grushton/develop/duos-ui/node_modules/paging-algorithm/src/index.ts' file: Error: ENOENT: no such file or directory, open '/Users/grushton/develop/duos-ui/node_modules/paging-algorithm/src/index.ts'
> ```

Yeah I don't have control over that. Basically webpack can't parse the source-map of a missing file in `paging-algorithm`, which is a deep dependency of `react-pagination` (old code strikes again). Obviously I can't supply a missing file to an internal library. That said, there shouldn't be any adverse effects for functionality, we just won't be able to read the un-minified code for that library in devTools while exploring/debugging the app.

**Review by quazi-h — APPROVED** [review]
Removal of PDF components looks good. Tested locally and I no longer see the download link.

File: src/index.js [hunk 1]
```diff
@@ -7,16 +7,14 @@ import 'jquery/src/jquery';
...
 
 import React from 'react';
-import ReactDOM from 'react-dom';
+import { createRoot } from 'react-dom/client';
 import './index.css';
 import App from './App';
 import { unregister } from './registerServiceWorker';
 import { BrowserRouter } from 'react-router-dom';
 
 unregister();
```
Comment (quazi-h) [inline src/index.js:17]: Actually I did notice an error in my terminal - it says compiled with 1 warning:
`Failed to parse source map from '/Users/qhoque/GIT/duos-ui/node_modules/paging-algorithm/src/index.ts' file: Error: ENOENT: no such file or directory, open '/Users/qhoque/GIT/duos-ui/node_modules/paging-algorithm/src/index.ts'

WARNING in ./node_modules/paging-algorithm/dist/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/qhoque/GIT/duos-ui/node_modules/paging-algorithm/src/index.ts' file: Error: ENOENT: no such file or directory, open '/Users/qhoque/GIT/duos-ui/node_modules/paging-algorithm/src/index.ts'`

I did not see the same warning when running locally from develop.

Comment (JVThomas) [reply src/index.js:17]: I've left a comment on it above

**Pull request merged**